File: docs/snippets/all/concepts/file_sink.cpp
// Create and set a file sink.

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_file_sink");

    rec.set_sinks(dalaran::FileSink{"recording.dlr"}).exit_on_failure();
}

File: docs/snippets/all/concepts/file_sink.rs
//! Create and set a file sink.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = dalaran::RecordingStreamBuilder::new("dalaran_example_file_sink")
        .buffered()?;

    rec.set_sink(Box::new(dalaran::sink::FileSink::new("recording.dlr")?));

    Ok(())
}
